pkg/util/mount/mount.go

```diff
@@ -76,3 +80,40 @@ func GetMountRefs(mounter Interface, mountPath string) ([]string, error) {
 	}
 	return refs, nil
 }
+
+// GetDeviceRefCount: given a device path, find its reference count from /proc/mounts
+// returns the reference count to the device and error code
+func GetDeviceRefCount(mounter Interface, deviceName string) (int, error) {
+	mps, err := mounter.List()
+	if err != nil {
+		return -1, err
+	}
+
+	// Find the number of references to the device.
+	refCount := 0
+	for i := range mps {
+		if strings.HasPrefix(mps[i].Device, deviceName) {
+			refCount++
+		}
+	}
+	return refCount, nil
+}
+
+// GetDeviceFromMnt: given a mnt point, find the device from /proc/mounts
+// returns the device name, reference count, and error code
+func GetDeviceFromMnt(mounter Interface, mnt string) (string, int, error) {
+	mps, err := mounter.List()
+	if err != nil {
+		return "", -1, err
+	}
+	device := ""
+	// Find the number of references to the device.
+	refCount := 0
+	for i := range mps {
+		if mps[i].Path == mnt {
+			device = mps[i].Device
```

Comment:
the issue of seeing multiple devices mounted on the same path is rather complicated and other APIs (e.g. <code>mount.GetMountRefs</code>) are also vulnerable to this problem.

on the other hand, the mount paths are constructed to be unique on a node, so the chance of collision is essentially non-existent. 

the worst case scenario is that the device keeps mounted upon pod cleanup, if it shares the same mount point as others. In such case, a dedicated cleaner is better suited for such task.